pkg/trace/api/api.go

```diff
@@ -408,7 +408,8 @@ func (r *HTTPReceiver) replyOK(v Version, w http.ResponseWriter) {
 	case v01, v02, v03:
 		httpOK(w)
 	default:
-		httpRateByService(w, r.dynConf)
+		n := httpRateByService(w, r.dynConf)
```

Comment:
I believe this value should only be used if no errors occurred while writing the response. The value does not represent what we're after in presence of errors.

Reply:
You are right. I will add a secondary `ok` return value which will indicate if the operation was successful. I think that should be sufficient.

Reply:
Please check the latest commit. Let me know if you're happy with the outcome.